src/state_transition/utils/process_proposer_lookahead.zig

```diff
@@ -4,129 +4,68 @@ const ssz = @import("consensus_types");
-    defer active_indices_list.deinit();
-
-    var seed: [32]u8 = undefined;
-    try getSeed(state, epoch, c.DOMAIN_BEACON_PROPOSER, &seed);
-
-    const proposer_indices = try allocator.alloc(ValidatorIndex, preset.SLOTS_PER_EPOCH);
-    try computeProposerIndices(
-        allocator,
-        epoch,
-        &seed,
-        active_indices_list.items,
-        effective_balance_increments,
-        proposer_indices,
-    );
-
-    return proposer_indices;
-}
+const seed_utils = @import("./seed.zig");
+const getSeed = seed_utils.getSeed;
+const computeProposers = seed_utils.computeProposers;
 
 /// Initializes `proposer_lookahead` during the Electra -> Fulu upgrade.
 /// Fills the `proposer_lookahead` field with `(MIN_SEED_LOOKAHEAD + 1)` epochs worth of proposer indices.
+/// Uses active indices from the epoch cache shufflings.
 pub fn initializeProposerLookahead(
     allocator: Allocator,
-    state: *const BeaconStateAllForks,
-    effective_balance_increments: *const EffectiveBalanceIncrements,
+    cached_state: *const CachedBeaconStateAllForks,
     out: []ValidatorIndex,
 ) !void {
-    std.debug.assert(effective_balance_increments.items.len > 0);
+    const epoch_cache = cached_state.epoch_cache_ref.get();
+    const state = cached_state.state;
+
     const lookahead_epochs = preset.MIN_SEED_LOOKAHEAD + 1;
     const expected_len = lookahead_epochs * preset.SLOTS_PER_EPOCH;
     std.debug.assert(out.len == expected_len);
 
     const current_epoch = computeEpochAtSlot(state.slot());
+    const effective_balance_increments = epoch_cache.getEffectiveBalanceIncrements();
+    const fork_seq = state.forkSeq();
 
     // Fill proposer_lookahead with current epoch through current_epoch + MIN_SEED_LOOKAHEAD
     for (0..lookahead_epochs) |i| {
         const epoch = current_epoch + i;
-        const epoch_proposers = try getBeaconProposerIndices(
+        const offset = i * preset.SLOTS_PER_EPOCH;
+
+        // Get active indices from the epoch cache
+        const active_indices = epoch_cache.getActiveIndicesAtEpoch(epoch) orelse return error.ActiveIndicesNotFound;
+
+        var seed: [32]u8 = undefined;
+        try getSeed(state, epoch, c.DOMAIN_BEACON_PROPOSER, &seed);
+
+        try computeProposers(
             allocator,
-            state,
+            fork_seq,
+            seed,
             epoch,
+            active_indices,
             effective_balance_increments,
+            out[offset .. offset + preset.SLOTS_PER_EPOCH],
         );
-        defer allocator.free(epoch_proposers);
-
-        const offset = i * preset.SLOTS_PER_EPOCH;
-        std.mem.copyForwards(ValidatorIndex, out[offset .. offset + preset.SLOTS_PER_EPOCH], epoch_proposers);
     }
 }
 
 /// Updates `proposer_lookahead` during epoch processing.
 /// Shifts out the oldest epoch and appends the new epoch at the end.
+/// Uses active indices from the epoch transition cache for the new epoch.
 pub fn processProposerLookahead(
```

Comment:
would be nice to move this function to under `src/state_transition/epoch`, similar to what ts implementation does